java.util.ConcurrentModificationException (no error message)

I think there is a bug with the realm plugin, when task: :compileKotlinIosSimulatorArm64 is running, i am getting the following:

```
* What went wrong:
Execution failed for task ':compileKotlinIosSimulatorArm64'.
> java.util.ConcurrentModificationException (no error message)
```

Sometimes it works and sometimes it doesn't, i guess due to the concurrency.

```
i: [Realm Compiler Plugin] 2022-01-07T09:43:13.830916Z Processing class Alias

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.831753Z String property named stringValue is nullable false

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836332Z Processing class Category

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836418Z Object property named wallpaper is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836575Z String property named title is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836598Z String property named trackingName is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836614Z String property named path is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836630Z Boolean property named isHidden is nullable false

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836649Z RealmList property named cells is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836971Z RealmList property named showcases is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.836995Z RealmList property named sections is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837015Z RealmList property named aliases is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837032Z Int property named rawCategoryType is nullable false

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837587Z Processing class Cell

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837606Z String property named id is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837627Z Int property named rawType is nullable false

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837910Z Processing class Section

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837930Z Int property named size is nullable false

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837959Z Boolean property named showTitle is nullable false

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837978Z String property named title is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.837991Z String property named viewAllUrl is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838002Z String property named trackingName is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838029Z RealmList property named cells is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838349Z Processing class Wallpaper

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838378Z String property named rawHighlightColour is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838399Z String property named rawBackgroundColour is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838419Z Int property named rawTheme is nullable false

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838443Z String property named topImage is nullable true

i: [Realm Compiler Plugin] 2022-01-07T09:43:13.838455Z String property named bottomImage is nullable true
```

Here is the same with --stacktrace
```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':compileKotlinIos'.
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:94)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
Caused by: java.util.ConcurrentModificationException
	at org.gradle.api.internal.DefaultDomainObjectCollection$IteratorImpl.next(DefaultDomainObjectCollection.java:475)
	at io.realm.gradle.RealmAnalytics.jsonPayload(RealmAnalytics.kt:112)
	at io.realm.gradle.RealmAnalytics.sendMetricIfNeeded(RealmAnalytics.kt:90)
	at io.realm.gradle.RealmAnalytics.afterExecute(RealmAnalytics.kt:82)
	at jdk.internal.reflect.GeneratedMethodAccessor184.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:464)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:446)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:61)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast$ListenerDispatch.dispatch(DefaultListenerManager.java:434)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast$ListenerDispatch.dispatch(DefaultListenerManager.java:421)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:43)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:245)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:157)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:141)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:37)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy89.afterExecute(Unknown Source)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:91)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,27 @@
+## 0.8.1 (2022-01-17)
+
+### Breaking Changes
+* None.
+
+### Enhancements
+* None.
+
+### Fixed
+* Synchronously process project configurations to avoid exceptions when running parallel builds [#626](https://github.com/realm/realm-kotlin/issues/626).
+
+### Compatibility
+* This release is compatible with:
+  * Kotlin 1.6.10.
+  * Coroutines 1.5.2-native-mt.
+  * AtomicFu 0.17.0.
+* Minimum Gradle version: 6.1.1.  
+* Minimum Android Gradle Plugin version: 4.0.0.
+* Minimum Android SDK: 16.
+
+### Internal
+* None.
+
+
 ## 0.8.0 (2021-12-17)
 
 ### Breaking Changes

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,4 @@
-## 0.8.1 (2022-01-17)
+## 0.8.1-SNAPSHOT (YYYY-MM-DD)
 
 ### Breaking Changes
 * None.

--- a/packages/gradle-plugin/src/main/kotlin/io/realm/gradle/RealmAnalytics.kt
+++ b/packages/gradle-plugin/src/main/kotlin/io/realm/gradle/RealmAnalytics.kt
@@ -83,6 +83,7 @@ internal class RealmAnalytics : TaskExecutionAdapter() {
         }
     }
 
+    @Synchronized
     private fun sendMetricIfNeeded(project: Project) {
         if (!METRIC_PROCESSED) {
             val disableAnalytics: Boolean = project.gradle.startParameter.isOffline || "true".equals(System.getenv()["REALM_DISABLE_ANALYTICS"], ignoreCase = true)

